src/mpid/ch4/src/ch4_progress.h

```diff
@@ -14,21 +14,26 @@
 cvars:
     - name        : MPIR_CVAR_CH4_GLOBAL_PROGRESS
       category    : CH4
-      type        : boolean
-      default     : 1
+      type        : enum
+      default     : normal
       class       : none
       verbosity   : MPI_T_VERBOSITY_USER_BASIC
       scope       : MPI_T_SCOPE_LOCAL
-      description : >-
-        If on, poll global progress every once a while. With per-vci configuration, turning global progress off may improve the threading performance.
+      description : |-
+        Sets the frequency of polling global progress. It is only relevant when multiple VCIs are enabled.
+        low - only poll global progress very infrequently
+        normal - the default frequency for typical applications
+        high - always poll global progress
 
 === END_MPI_T_CVAR_INFO_BLOCK ===
 */
 
 /* Global progress (polling every vci) is required for correctness. Currently we adopt the
  * simple approach to do global progress every MPIDI_CH4_PROG_POLL_MASK.
  */
-#define MPIDI_CH4_PROG_POLL_MASK 0xff
+#define MPIDI_CH4_PROG_POLL_NORMAL_MASK 0xff    /* 256 x 1us = 1/4 ms */
+#define MPIDI_CH4_PROG_POLL_LOW_MASK 0xfffffff  /* 268435456 x 1us = 4.5 min. */
```

Comment:
What would happen in a scenario such as: an application is ready to finalize + terminate but the current global progress interval is not completed? E.g., there are still 4 more mins until the next global progress?
Few possibilities that I can quickly think of are: Finalize will invoke an early global progress, or the application will stall until the next global progress.

Reply:
The LOW frequency is effectively turning global progress off, so assumably user only turn it on when they know their application doesn't not need global progress. Otherwise, the long progress stall is their fault -- at least it is not a deadlock :)

Reply:
Oh, that's actually right. LOW frequency is almost equivalent to turning it off, with possible far away global progress every few minutes.